tests/file_storage/tests.py

```diff
@@ -123,6 +128,112 @@ def test_file_access_options(self):
         self.storage.delete('storage_test')
         self.assertFalse(self.storage.exists('storage_test'))
 
+    def _test_file_time_getter(self, getter):
+        # We want to check for correct behaviour under both
+        # USE_TZ=True and USE_TZ=False. The tests are
+        # annoyingly similar, since they both set up a
+        # situation where system, Django and UTC are distinct.
+        self._test_file_time_getter_tz_handling_on(getter)
+        self._test_file_time_getter_tz_handling_off(getter)
+
+    @override_settings(USE_TZ=True, TIME_ZONE='Africa/Algiers')
+    def _test_file_time_getter_tz_handling_on(self, getter):
+        # We've set the Django TZ (and hence the system TZ) to
+        # Africa/Algiers, which is UTC+1 and has no DST change.
+        # Then we can set the Django TZ to something else so
+        # UTC, Django and system are all reliably different.
+
+        now_in_algiers = timezone.make_aware(datetime.now())
+
+        # Use a fixed offset timezone so we don't need pytz.
+        timezone.activate(timezone.get_fixed_timezone(-300))
```

Comment:
I'll take this approach, because I want to get this through, but I still don't think it's actually better than using `try…finally`. (Language features over library functions, in my book.)